src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonPropertyInfoOfT.cs

```diff
@@ -500,24 +423,17 @@ internal override void SetExtensionDictionaryAsObject(object obj, object? extens
                     goto case JsonIgnoreCondition.WhenWritingDefault;
                 case JsonIgnoreCondition.WhenWritingDefault:
                     {
-                        if (_propertyTypeEqualsTypeToConvert)
-                        {
-                            return ShouldSerializeIgnoreConditionWhenWritingDefaultPropertyTypeEqualsTypeToConvert;
-                        }
-                        else
-                        {
-                            return ShouldSerializeIgnoreConditionWhenWritingDefaultPropertyTypeNotEqualsTypeToConvert;
-                        }
```

Comment:
splitting this was meant to remove extra check during serialization - rather than checking variable on every Serialize we check it once here and then remove check in the method